src/ngraph/pass/implicit_broadcast_elimination.hpp

```diff
@@ -0,0 +1,49 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/op/util/broadcasting.hpp"
+#include "ngraph/pass/pass.hpp"
+
+namespace ngraph
+{
+    namespace pass
+    {
+        template <typename T>
+        NodeVector static explicit_broadcast(std::shared_ptr<T>& node)
+        {
+            if (node->get_autob().type == op::AutoBroadcastType::NONE)
+            {
+                return node->get_arguments();
```

Comment:
by having multiple returns you do not allow the compiler to use Return Value Optimization, forcing it to make a copy when returning NodeVector.

Reply:
This is only used on nodes with two arguments. As such copying might not be too bad. Do you have an alternative in mind?

Reply:
at the top define `NodeVector rc;`
instead of returns set `rc = <whatever>;`
then at the bottom `return rc;`
One return so RVO works. It is a good general rule to follow to only have one return per function if you  are returning something non trivial